EventsExpress/Validation/ChangePasswordViewModelValidator.cs

```diff
@@ -1,15 +1,40 @@
-﻿using EventsExpress.ViewModels;
+﻿using System.Linq;
+using EventsExpress.Core.IServices;
+using EventsExpress.Db.Bridge;
+using EventsExpress.Db.EF;
+using EventsExpress.ViewModels;
 using FluentValidation;
+using Microsoft.EntityFrameworkCore;
 
 namespace EventsExpress.Validation
 {
     public class ChangePasswordViewModelValidator : AbstractValidator<ChangeViewModel>
     {
-        public ChangePasswordViewModelValidator()
+        private readonly AppDbContext appDbContext;
+        private readonly IPasswordHasher passwordHasher;
+        private readonly ISecurityContext securityContext;
+
+        public ChangePasswordViewModelValidator(AppDbContext appDbContext, IPasswordHasher passwordHasher, ISecurityContext securityContext, IAuthService authService)
         {
+            this.appDbContext = appDbContext;
+            this.passwordHasher = passwordHasher;
+            this.securityContext = securityContext;
+
             RuleFor(x => x.OldPassword).NotEmpty().WithMessage("Password is required");
+            RuleFor(x => x.OldPassword).Must(ValidOldPassword).WithMessage("Wrong password");
             RuleFor(x => x.NewPassword).NotEmpty().WithMessage("Password is required");
             RuleFor(x => x.NewPassword).NotEqual(x => x.OldPassword).WithMessage("New Password can't be the same as Old Password!");
         }
+
+        protected bool ValidOldPassword(string password)
+        {
+            var userDto = appDbContext.Users
```

Comment:
I guess we do not need the whole user object here. You can get `AccountId` through `securityContext` and try call `appDbContext.AuthLocal.First(by AccountId)` directly

Reply:
I thought of that as well. But I believe we do not have accountid in claims. That would be a different activity.